### ISSUE TITLE
Enable batch video upload specific to a dataset

### ISSUE THREAD

**Issue by ocrasborn** [Signbank/Global-signbank#388]
Two alternative ideas:
* Force users to prefix their video file names with the dataset name
* Create a video upload folder on the server for each dataset
Even then, we'd need to specify somewhere for a given dataset which gloss field it tries to match to the file names of the videos. Right now it's the Annotation ID Gloss: Dutch, but for some others (like Shanghainese) we would rather choose for the English gloss in the filenames.

**Comment by henrinie** [comment]
FinSL-signbank just has a form that you can use to upload multiple files, and then a view in which you can link the videos to glosses. When uploading (multiple) videos you will have to select a dataset.

**Comment by ocrasborn** [comment]
That's also a good idea. Workflow-wise I like the idea that people prepare file names in accordance with the glosses though, which would enable skipping the view. In the upload form, people would have to both specify the data set and the gloss field which is matched to the file names.

**Comment by vanlummelhuizen** [comment]
For now I have made a quick workaround. From the documentation of import_media() in dictionary/views.py:

> Importing media is done as follows:
> 
> 1. In the main folder for importing media all folders with names equal to dataset names are opened.
> 2. In each folder with a dataset name all folders with name equal to a three letter language code of the dataset at hand are opened.
> 3. In the folders with a three letter language code as name all media file are imported.
>     
> E.g. view the following folder structure:
>      <import folder>/ASL/eng/HOUSE.mp4
>      
> where 'ASL' is the name of a dataset and 'eng' is the three letter language code of English which is one of the dataset languages of the dataset 'ASL'.

**Comment by ocrasborn** [comment]
Hoi Micha,

Ik wilde meteen even een hele serie glossen aanmaken, maar liep vast: het import-script voor het aanmaken van glossen lijkt niet meer te werken. Bijgevoegd het gebruikte bestand. Kun je hier nog naar kijken? Dan test ik in parallel de video-upload met wat nepgebaren.

Groet,
Onno

**Comment by vanlummelhuizen** [comment]
@ocrasborn waar is het bestand?

**Comment by ocrasborn** [comment]
Quickly tested: created two glosses ASK and ASK-1, uploaded two .mov files, but the script returns an error.

**Comment by ocrasborn** [comment]
Feel free to test it live: all the videos for the current Shanghainese glosses are in the right folder. (Next week, that is, no need to work on it over the weekend.)

**Comment by vanlummelhuizen** [comment]
@ocrasborn , there was a small bug still. It now works om my local machine. Will you do the honors? 

Because you have uploaded .mov files, they are not displayed on the Gloss detail view, since this was tailored to mp4.

**Comment by ocrasborn** [comment]
Hi @vanlummelhuizen : It worked with two test files that were still there, but now with the rest of the batch I get a new error:
"local variable 'gloss' referenced before assignment".

**Comment by ocrasborn** [comment]
Thanks @vanlummelhuizen , all goes well now! Would you mind updating the wiki page on how to batch import videos, or have you already done so?

**Comment by vanlummelhuizen** [comment]
@ocrasborn just updated the wiki. 

I saw you changed the .mov extension to .mp4 for the uploaded videos. This does work on Mac (also in Chrome) but not on my Ubuntu with Firefox. Any idea how it behave on Windows?

In the end, a QuickTime Movie is not MPEG4, so we should solve this is more constructive way I think.

**Comment by vanlummelhuizen** [comment]
@ocrasborn it also works on Firefox on Ubuntu Linux.

**Comment by vanlummelhuizen** [comment]
@ocrasborn close?

**Comment by ocrasborn** [comment]
yes indeed @vanlummelhuizen
